webapp/src/components/link_tooltip/link_tooltip.jsx

```diff
@@ -0,0 +1,170 @@
+                    <div className='header mb-1'>
+                        <a
+                            title={data.repo}
+                            href={href}
+                        >
+                            {data.repo}
+                        </a>
+                        &nbsp;on&nbsp;<span>{date}</span>
+                    </div>
+
+                    <div className='body d-flex mt-2'>
+                        <span className='pt-1 pb-1 pr-2'>
+                            {getIconElement()}
+                        </span>
+
+                        {/* info */}
+                        <div className='tooltip-info mt-1'>
+                            <a href={href}>
+                                <h5 className='mr-1'>{data.title}</h5>
+                                <span className='mr-number'>#{data.iid}</span>
+                            </a>
+                            <div className='markdown-text mt-1 mb-1'>
+                                <ReactMarkdown
+                                    source={data.description}
+                                    disallowedTypes={['heading']}
+                                    linkTarget='_blank'
+                                />
+                            </div>
+
+                            {/* base <- head */}
+                            {data.type === 'merge_requests' && (
+                                <div className='base-head mt-1 mr-3'>
+                                    <span
+                                        title={data.target_branch}
+                                        className='commit-ref'
+                                        style={{maxWidth: '140px'}}
+                                    >
+                                        {data.target_branch}
+                                    </span>
+                                    <span className='mx-1'>←</span>
+                                    <span
+                                        title={data.source_branch}
+                                        className='commit-ref'
+                                    >
+                                        {data.source_branch}
+                                    </span>
+                                </div>
+                            )}
+
+                            <div className='see-more mt-1'>
+                                <a
+                                    href={href}
+                                    target='_blank'
+                                    rel='noopener noreferrer'
+                                >
+                                    See more
+                                </a>
+                            </div>
+
+                            {/* Labels */}
+                            <div className='labels mt-3'>
+                                {data.labels && data.labels_with_details && data.labels_with_details.map((label, index) => {
```

Comment:
it is a good idea before running a map check for the length of the array too.

Reply:
Fixed.